openmc/stats/multivariate.py

```diff
@@ -613,6 +615,116 @@ def from_xml_element(cls, elem):
         origin = [float(x) for x in elem.get('origin').split()]
         return cls(r, phi, z, origin=origin)
 
+class MeshIndependent(Spatial):
+    """Spatial distribution for a mesh.
+
+    This distribution allows one to specify a mesh to sample over and the scheme to sample over the entire mesh. 
+
+    .. versionadded:: TODO
+
+    Parameters
+    ----------
+    elem_weight_scheme : String
+        The scheme for weighting and sampling elements from the mesh. Options are file, volume, and equal weights.
+    mesh_id : int, optional
+        The mesh ID number, defaults to 1
+    weights_file : Iterable of Real, optional
+        A list of values which represent the weights of each element
+
+
+    Attributes
+    ----------
+    elem_weight_scheme : String
+        Weighting scheme for sampling element from mesh
+    mesh_id : int, optional
+        The ID of the mesh, defaults to 1
+    weights_file : Iterable of Real, optional
+        A list of values which represent the weights of each element
+
+    """
+
+    def __init__(self, elem_weight_scheme, mesh_id=1, weights_file=[0.0]):
+        self.elem_weight_scheme = elem_weight_scheme
+        self.mesh_id = mesh_id
+        self.weights_file = weights_file
+
+    @property
+    def elem_weight_scheme(self):
+        return self._elem_weight_scheme
+
+    @elem_weight_scheme.setter
+    def elem_weight_scheme(self, elem_weight_scheme):
+        cv.check_type('Scheme for sampling an element from the mesh', elem_weight_scheme, str)
+        if elem_weight_scheme == 'volume' or elem_weight_scheme == 'file' or elem_weight_scheme == 'equal':
+            self._elem_weight_scheme = elem_weight_scheme
+        else:
+            raise Exception("Type of element sampling scheme provided is not in the supported types (volume, file, or equal)")
```

Comment:
I think we can do this same check with:

```suggestion
        cv.check_value('Sampling scheme', elem_weight_scheme, ('volume', 'file', 'equal')
```